test/controllers/customers_controller_test.rb

```diff
@@ -0,0 +1,48 @@
+require "test_helper"
+
+class CustomersControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    @customer = customers(:one)
+  end
+
+  test "should get index" do
+    get customers_url
+    assert_response :success
+  end
+
+  test "should get new" do
+    get new_customer_url
+    assert_response :success
+  end
+
+  test "should create customer" do
+    assert_difference("Customer.count") do
+      post customers_url, params: { customer: { email: @customer.email, first_name: @customer.first_name, last_name: @customer.last_name, phone: @customer.phone } }
+    end
+
+    assert_redirected_to customer_url(Customer.last)
+  end
+
+  test "should show customer" do
+    get customer_url(@customer)
+    assert_response :success
+  end
+
+  test "should get edit" do
+    get edit_customer_url(@customer)
+    assert_response :success
+  end
+
+  test "should update customer" do
+    patch customer_url(@customer), params: { customer: { email: @customer.email, first_name: @customer.first_name, last_name: @customer.last_name, phone: @customer.phone } }
```

Comment:
This test doesn't assert that the customer attributes changed and only tests if you are redirected to the right page. Ideally at least both of those things are tested for.